Closes #74: Add missing component flows

michaelheckmann opened the pull request:
# Add missing component flows

<img width="1920" height="1080" alt="11shots_so" src="https://github.com/user-attachments/assets/5bd1abb0-1e6d-44f0-ba27-ddca2af7a148" />

## :recycle: Current situation & Problem
The app is missing the health data and questionnaire pages.

## :gear: Release Notes

### New UI Components and Forms
-  **Duration Input**: Added `DurationInput.tsx` for flexible time entry (days/weeks with automatic unit switching for optimal display, e.g., "2 weeks" instead of "14 days"). Includes conversion logic, validation, and singular/plural formatting. Integrated into multiple forms for study duration, historical data, and schedule offsets.
-  **Health Data Form**: Created `HealthDataComponentForm.tsx` with MultiSelect for choosing sample types (categorized by health data types like Heart Rate, Sleep Analysis). Supports historical data collection via DurationInput and required validation (at least one sample type).
-  **Questionnaire Form**: New `QuestionnaireComponentForm.tsx` for FHIR JSON input with Textarea editor. Features upload/download buttons for JSON files (with validation and formatting), a "Format JSON" button (prettifies with 2-space indentation), and error handling for invalid JSON.

### Routing and Workflow Enhancements
-  **Component Creation/Editing Routes**: Updated `new.tsx` and `edit.tsx` to handle all component types (information, health-data, questionnaire) with conditional rendering of type-specific forms. Standardized prop names (e.g., `onSubmit` instead of `onSave`). Added navigation logic and hotkey support (Cmd+Enter).
-  **Edit Layout**: Modified `EditComponentLayout.tsx` to conditionally show/hide the ScheduleDialog button (disabled for health-data components, as they may not need scheduling). Includes delete functionality with navigation back to the components list.

## :white_check_mark: Testing
Extended `components.spec.ts` with tests for health data and questionnaire components.

## :pencil: Code of Conduct & Contributing Guidelines
By creating and submitting this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [X] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

michaelheckmann commented on src/routes/~(dashboard)/~$team/~$study/~configuration/~components/~new.tsx:
```diff
@@ -63,7 +65,13 @@ const NewComponentRoute = () => {
         <Card>
           {componentType === "information" && (
-            <InformationComponentForm form={form} onSave={handleSave} />
+            <InformationComponentForm form={form} onSubmit={handleSave} />
```
Done.

arkadiuszbachorski commented on src/routes/~(dashboard)/~$team/~$study/~configuration/~components/~new.tsx:
```diff
@@ -63,7 +65,13 @@ const NewComponentRoute = () => {
         <Card>
           {componentType === "information" && (
-            <InformationComponentForm form={form} onSave={handleSave} />
+            <InformationComponentForm form={form} onSubmit={handleSave} />
```
`handleSubmit` for consistency?

arkadiuszbachorski commented on src/routes/~(dashboard)/~$team/~$study/~configuration/~components/~$component.tsx:
```diff
@@ -59,7 +62,13 @@ const EditComponentRoute = () => {
         <Card>
           {componentType === "information" && (
-            <InformationComponentForm form={form} onSave={handleSave} />
+            <InformationComponentForm form={form} onSubmit={handleSave} />
```
`handleSubmit` for consistency?

arkadiuszbachorski commented on src/routes/~(dashboard)/~$team/~$study/~configuration/~components/components/HealthDataComponentForm.tsx:
```diff
@@ -0,0 +1,93 @@
+              </MultiSelectContent>
+            </MultiSelect>
+          )}
+          className="px-6 pt-6"
```
This is a nit, but I think it's nice to keep `render` as the most bottom part of the `Field` component. It can be easy to miss other props under `render` otherwise.

arkadiuszbachorski reviewed (changes requested):
I checked it locally and mostly works great, but found two things.

1. When I first choose "Questionnaire", try to submit it empty - it results with a validation error - missing JSON. Then, when I change type and try to submit - it still remembers the old validation error from "Questionnaire", preventing me from proper submission.

2. When clicking "Change type", it always pre-selects "Information" type. It would make sense to pre-select currently active type.

michaelheckmann commented:
Thanks for the review @arkadiuszbachorski !
Great catch! Fixed the first issue by running a `form.reset` when the `componentType` changes. Also adjusted the `NewComponentDialog` to preselect the current component type.
